main/src/library/File.flix

```diff
@@ -15,6 +15,176 @@
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns the last-modified timestamp of the given file `f` in milliseconds
+    ///
+    pub def modificationTime(f: String): Result[Int64, String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import static java.nio.file.Files.readAttributes(##java.nio.file.Path, ##java.lang.Class, Array[##java.nio.file.LinkOption]);
+            import static java.lang.Class.forName(String);
+            import java.nio.file.attribute.BasicFileAttributes.lastModifiedTime();
+            import java.nio.file.attribute.FileTime.toMillis();
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let classOfAttributes = forName("java.nio.file.attribute.BasicFileAttributes");
+            let fileAttributes = readAttributes(javaPath, classOfAttributes, []);
+            let timeLastModified = lastModifiedTime(fileAttributes);
+            let inMilliSec = toMillis(timeLastModified);
+
+            Ok(inMilliSec)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns the size of the given file `f` in bytes.
+    ///
+    pub def size(f: String): Result[Int64, String] & Impure =
+       try {
+           import new java.io.File(String) as newFile;
+           import java.io.File.toPath();
+           import static java.nio.file.Files.readAttributes(##java.nio.file.Path, ##java.lang.Class, Array[##java.nio.file.LinkOption]);
+           import static java.lang.Class.forName(String);
+           import java.nio.file.attribute.BasicFileAttributes.size();
+
+           let javaFile = newFile(f);
+           let javaPath = toPath(javaFile);
+           let classOfAttributes = forName("java.nio.file.attribute.BasicFileAttributes");
+           let fileAttributes = readAttributes(javaPath, classOfAttributes, []);
+           let fileSize = size(fileAttributes);
+
+           Ok(fileSize)
+       } catch {
+           case ex: ##java.io.IOException =>
+               import java.lang.Throwable.getMessage();
+               Err(getMessage(ex))
+       }
+
+    ///
+    /// Returns the type of the given file `f`.
+    ///
+    pub def fileType(f: String): Result[FileType, String] & Impure =
+        let isFile = File.isRegularFile(f);
+        let isDirec = File.isDirectory(f);
+        let isSymLink = File.isSymbolicLink(f);
+
+        match    (isFile  , isDirec  , isSymLink) {
+            case (Err(v)  , _        , _        ) => Err(v)
+            case (_       , Err(v)   , _        ) => Err(v)
+            case (_       , _        , Err(v)   ) => Err(v)
+            case (Ok(file), Ok(direc), Ok(sym)  ) => {
```

Comment:
Is this too much alignment?

Reply:
Its OK, but I think you should do it as a staircase instead. 

```
if (...)
  if (...)
    if (...)
```

is that not better?

Reply:
`let*` would probably be cleanest

Reply:
See here: https://doc.flix.dev/tipstricks/ for let*